components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/LoggingResource.java

```diff
@@ -0,0 +1,163 @@
+                        loggerName = jsonPayload.getString(Constants.LOGGER_NAME);
+                        if(loggerName.equals("root")){
+                            // update root logger
+                            jsonBody = updateSystemLog(logLevel);
+                        } else {
+                            // update the specific logger
+                            jsonBody = updateLoggerData(axis2MessageContext, loggerName, logLevel);
+                        }
+                    } else {
+                        // 400-Bad Request logger name is missing
+                        jsonBody = Utils.createJsonErrorObject("Logger name is missing");
+                        axis2MessageContext.setProperty(Constants.HTTP_STATUS_CODE, Constants.BAD_REQUEST);
+                    }
+                }
+            } else {
+                // 400-Bad Request loggingLevel is missing
+                jsonBody = Utils.createJsonErrorObject("Logging level is missing");
+                axis2MessageContext.setProperty(Constants.HTTP_STATUS_CODE, Constants.BAD_REQUEST);
+            }
+        }
+        Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+        return true;
+    }
+
+    private JSONObject updateSystemLog(String logLevel) {
+
+        JSONObject jsonBody = new JSONObject();
+
+        // update root logger details
+        Logger rootLogger = Logger.getRootLogger();
+        rootLogger.setLevel(Level.toLevel(logLevel));
+
+        // update logger and appender data, following are set
+        Enumeration loggersEnum = LogManager.getCurrentLoggers();
+        Level systemLevel = Level.toLevel(logLevel);
+        while (loggersEnum.hasMoreElements()) {
+            Logger logger = (Logger) loggersEnum.nextElement();
+            logger.setLevel(systemLevel);
+        }
+        jsonBody.put(Constants.MESSAGE, "Successfully updated system log level to " + rootLogger.getLevel().toString());
+        return jsonBody;
+    }
+
+    private JSONObject updateLoggerData(org.apache.axis2.context.MessageContext axis2MessageContext, String loggerName, String loggerLevel) {
+
+        JSONObject jsonBody = new JSONObject();
+        //update logger data in current system
+        Logger logger = LogManager.exists(loggerName);
+
+        if (Objects.nonNull(logger)) {
+            logger.setLevel(Level.toLevel(loggerLevel));
+            jsonBody.put(Constants.MESSAGE, "Successfully updated " + loggerName + " level to " + logger.getLevel().toString());
+        } else {
+            jsonBody = Utils.createJsonErrorObject("Invalid logger " + loggerName);
+            axis2MessageContext.setProperty(Constants.HTTP_STATUS_CODE, Constants.BAD_REQUEST);
+        }
+        return jsonBody;
+    }
+
+    private JSONObject getLoggerData(org.apache.axis2.context.MessageContext axis2MessageContext, String loggerName) {
+
+        JSONObject jsonBody = new JSONObject();
+
+        if (loggerName.equals("root")) {
```

Comment:
Shall we make `"root"` a constant and use the value `"ROOT_LOGGER"` instead?

Reply:
Fixed in 5f84b6a